### PR TITLE
Add vue-loader and template support

### PR DESCRIPTION
**Pull request by OSpoon**
I added the necessary modules `vue-loader`, `ts-loader`, and `@vue/compiler-sfc` in `programs/develop`. Additionally, I have added vue and vue-typescript templates. 

Now, using the vue and vue-typescript templates, we can achieve the same presentation effect as with the react and react-typescript templates.

When you have time, please give me some suggestions. I would be very happy.

### CODE REVIEW

File: programs/create/templates/vue-typescript/template/content/ContentApp.vue
Comment (cezaraugusto) [inline programs/create/templates/vue-typescript/template/content/ContentApp.vue:1]: When I try to edit this file I get the following error in the console, and the content_script doesn't reload after changes. Do you have any hints?

Screenshot 2024-05-16 at 16 22 23

**Review by cezaraugusto — CHANGES_REQUESTED** [review]
Hi @OSpoon, this is a huge progress towards Vue support, thanks a lot! I got able to create and start the project using both templates! 🧩

There are a few changes I left in the comments. I also need you to please update the dev-server in [this line](https://github.com/cezaraugusto/extension.js/blob/main/programs/develop/webpack/webpack-dev-server.ts#L30) to set watchFiles to `undefined` not only for React but also Vue extensions. This way we can get true hot-module replacement for `.vue` files.

Hot-module replacement isn't working for `content_scripts`, unfortunately (using `vue-typescript` template).  It is not a blocker to land these changes and have at least partial support for Vue, but if you any existing config that managed to do this, I'd love to know and find ways to add it to a future patch.

File: programs/develop/webpack/loaders/vueLoaders.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    vueLoaders.push({
+      test: /\.vue$/,
+      loader: 'vue-loader'
+    })
+  }
+
+  if (isUsingTypeScript(projectDir)) {
+    vueLoaders.push({
+      test: /\.ts$/,
+      exclude: /background\.ts/,
```
[the remark this answers is not in the document]
Comment (OSpoon) [reply programs/develop/webpack/loaders/vueLoaders.ts:24]: Just like the error you encountered, it was still exposed during the hot update phase, but this error seemed to be just a type inconsistency problem, so I decided to enable the [transpileonly](https://github.com/TypeStrong/ts-loader#transpileonly) option to skip type checking.

Although transpileOnly will default to true after using [fork-ts-checker-webpack-plugin](https://github.com/TypeStrong/fork-ts-checker-webpack-plugin), it still seems to need to be turned on explicitly.

File: programs/create/templates/vue-typescript/template/content/ContentApp.vue
Comment (OSpoon) [reply programs/create/templates/vue-typescript/template/content/ContentApp.vue:1]: I considered explicitly enabling [transpileonly](https://github.com/TypeStrong/ts-loader#transpileonly) to skip type checking.

File: programs/develop/webpack/loaders/vueLoaders.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import {isUsingTypeScript} from '../options/typescript'
+import {isUsingVue} from '../options/vue'
+
+export default function vueLoaders(projectDir: string) {
+  const vueLoaders = []
+
+  if (isUsingVue(projectDir)) {
+    vueLoaders.push({
+      test: /\.vue$/,
+      loader: 'vue-loader'
```
[the remark this answers is not in the document]
Comment (OSpoon) [reply programs/develop/webpack/loaders/vueLoaders.ts:17]: I agree.

File: programs/develop/webpack/loaders/vueLoaders.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    })
+  }
+
+  if (isUsingTypeScript(projectDir)) {
+    vueLoaders.push({
+      test: /\.ts$/,
+      exclude: /background\.ts/,
+      loader: 'ts-loader',
```
[the remark this answers is not in the document]
Comment (OSpoon) [reply programs/develop/webpack/loaders/vueLoaders.ts:25]: I agree.

File: programs/develop/webpack/loaders/vueLoaders.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  if (isUsingVue(projectDir)) {
+    vueLoaders.push({
+      test: /\.vue$/,
+      loader: 'vue-loader'
+    })
+  }
+
+  if (isUsingTypeScript(projectDir)) {
+    vueLoaders.push({
```
[the remark this answers is not in the document]
Comment (OSpoon) [reply programs/develop/webpack/loaders/vueLoaders.ts:22]: I will narrow the scope of influence.

File: programs/create/templates/vue-typescript/template/content/content/ContentApp.vue
[the remark this answers is not in the document]
Comment (OSpoon) [reply programs/create/templates/vue-typescript/template/content/content/ContentApp.vue:1]: I agree.

File: programs/develop/webpack/webpack-config.ts [hunk 5]
```diff
@@ -95,7 +96,8 @@ export default function webpackConfig(
       rules: [
         ...jsLoaders(projectPath, devOptions),
         ...styleLoaders(projectPath, devOptions),
-        ...assetLoaders
+        ...assetLoaders,
+        ...vueLoaders(projectPath),
```
[the remark this answers is not in the document]
Comment (OSpoon) [reply programs/develop/webpack/webpack-config.ts]: I agree.

File: programs/develop/webpack/loaders/vueLoaders.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (OSpoon) [reply programs/develop/webpack/loaders/vueLoaders.ts]: In [cli-plugin-typescript/index.js](https://github.com/vuejs/vue-cli/blob/dev/packages/%40vue/cli-plugin-typescript/index.js#L64C9-L64C22) has the same configuration.

[the remark this answers is not in the document]
Comment (cezaraugusto) [reply programs/develop/webpack/loaders/vueLoaders.ts]: sounds good

**Review by cezaraugusto — APPROVED** [review]
LGTM ++

Thanks a lot for your contribution, @OSpoon! Will ensure this is added in the next release.

**Pull request merged**